sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/cpu/CpuReader.java

```diff
@@ -0,0 +1,50 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.cosmos.implementation.cpu;
+
+import com.azure.cosmos.implementation.Utils;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.lang.management.ManagementFactory;
+
+public class CpuReader {
+    private final static Logger logger = LoggerFactory.getLogger(CpuReader.class);
+    private final com.sun.management.OperatingSystemMXBean operatingSystemMXBean;
+
+    public CpuReader() {
+        java.lang.management.OperatingSystemMXBean mxBean = null;
+        try {
+            mxBean =
+                ManagementFactory.getOperatingSystemMXBean();
+        } catch (Throwable t) {
+            logger.error("failed to initialized CpuReader", t);
+        }
+
+        this.operatingSystemMXBean = tryGetAs(mxBean,
+            com.sun.management.OperatingSystemMXBean.class);
+    }
+
+    public double getSystemWideCpuUsage() {
+        try {
+            if (operatingSystemMXBean != null) {
+                return operatingSystemMXBean.getSystemCpuLoad();
```

Comment:
Negative value should also be mapped to Double.NaN ?

Reply:
correct. thanks for catching this. will fix.

Reply:
addressed the bug. thanks for catching it.